openerp/osv/fields.py

```diff
@@ -584,35 +585,69 @@ class binary(_column):
     _symbol_set = (_symbol_c, _symbol_f)
     _symbol_get = lambda self, x: x and str(x)
 
-    __slots__ = ['filters']
+    __slots__ = ['attachment', 'filters']
 
     def __init__(self, string='unknown', filters=None, **args):
         args['_prefetch'] = args.get('_prefetch', False)
+        args['attachment'] = args.get('attachment', True)
         _column.__init__(self, string=string, filters=filters, **args)
 
+    def to_field_args(self):
+        args = super(binary, self).to_field_args()
+        args['attachment'] = self.attachment
+        return args
+
     def get(self, cr, obj, ids, name, user=None, context=None, values=None):
-        if not context:
-            context = {}
-        if not values:
-            values = []
-        res = {}
-        for i in ids:
-            val = None
-            for v in values:
-                if v['id'] == i:
-                    val = v[name]
-                    break
-
-            # If client is requesting only the size of the field, we return it instead
-            # of the content. Presumably a separate request will be done to read the actual
-            # content if it's needed at some point.
-            # TODO: after 6.0 we should consider returning a dict with size and content instead of
-            #       having an implicit convention for the value
-            if val and context.get('bin_size_%s' % name, context.get('bin_size')):
-                res[i] = tools.human_size(long(val))
+        result = dict.fromkeys(ids, False)
+
+        if self.attachment:
+            # values are stored in attachments, retrieve them
+            atts = obj.pool['ir.attachment'].browse(cr, user, [], context)
+            domain = [
+                ('res_model', '=', obj._name),
+                ('res_field', '=', name),
+                ('res_id', 'in', ids),
+            ]
+            for att in atts.search(domain):
+                # the 'bin_size' flag is handled by the field 'datas' itself
+                result[att.res_id] = att.datas
+        else:
```

Comment:
`att` has the `bin_size` flag in its own context, therefore `att.datas` already handles it